src/common/src/types/mod.rs

```diff
@@ -570,10 +570,17 @@ pub trait ToOwnedDatum {
     fn to_owned_datum(self) -> Datum;
 }
 
-impl ToOwnedDatum for DatumRef<'_> {
+impl<T: Into<ScalarImpl>> ToOwnedDatum for T {
```

Comment:
TBH I'd prefer introducing a new trait for doing such conversion. By `ToOwnedDatum` I suppose it's only for the conversion from the ref type to an owned type, just like the std's `ToOwned` trait with GAT support.

Reply:
But logically types `impl Into<ScalarImpl>` can be converted `ToOwnedDatum`. I think it would be convenient to reuse this trait.